internal/pagefile_test.go

```diff
@@ -137,26 +327,75 @@ func deepCompare(file1, file2 string) bool {
 	}
 }
 
-func readIncrementFileHeaderTest(t *testing.T, headerData []byte, expectedErr error) {
-	err := internal.ReadIncrementFileHeader(bytes.NewReader(headerData))
-	assert.IsType(t, err, expectedErr)
+// TestIncrement holds information about some increment for easy testing
+type TestIncrement struct {
+	incrementBytes []byte
+	fileSize uint64
+	diffBlockCount uint32
 }
 
-func TestReadIncrementFileHeader_Valid(t *testing.T) {
-	readIncrementFileHeaderTest(t, internal.IncrementFileHeader, nil)
+func (ti *TestIncrement) NewReader() io.Reader {
+	return bytes.NewReader(ti.incrementBytes)
 }
 
-func TestReadIncrementFileHeader_InvalidIncrementFileHeaderError(t *testing.T) {
-	dataArray := [][]byte{
-		{'w', 'i', '1', 0x56},
-		{'x', 'i', '1', internal.SignatureMagicNumber},
-		{'w', 'j', '1', internal.SignatureMagicNumber},
+func newTestIncrement(lsn uint64) *TestIncrement {
+	incrementBytes := readIncrementToBuffer(lsn)
+	fileSize, diffBlockCount, _, _ := internal.GetIncrementHeaderFields(bytes.NewReader(incrementBytes))
+	return &TestIncrement{incrementBytes: incrementBytes, fileSize: fileSize, diffBlockCount: diffBlockCount}
+}
+
+// MockReadWriterAt used for mocking file in tests
+type MockReadWriterAt struct {
+	size int64
+	bytesWritten int
+	content      []byte
+}
+
+func NewMockReaderAtWriterAt(content []byte) *MockReadWriterAt {
```

Comment:
NewMockReadWriterAt

Reply:
Fixed in ed861e47a77988113cb0d21f7b58b6c483779efe